### PR TITLE
Take the gray out of gradients

### PR DESCRIPTION
**Pull request by jkbelcher**
Thanks to upstream assistance from mcslee, future LX releases will have improved color blending especially when blending to/from black.

This PR brings in the new blend functions so we can use them immediately.  This is a temporary patch until we upgrade to a version of LX that includes the new features.

image

### CODE REVIEW

File: src/main/java/titanicsend/lx/LXGradientUtils.java [hunk 1]
```diff
@@ -0,0 +1,318 @@
...
+      this.hue = color.getHuef() + hueOffset;
+      this.saturation = color.getSaturation();
+      this.brightness = LXColor.b(c);
+      setRGB(LXColor.hsb(this.hue, this.saturation, this.brightness));
+    }
+
+    public void set(LXDynamicColor color, float hueOffset, float saturationOffset, float brightnessOffset) {
+      int c = color.getColor();
+      this.hue = color.getHuef() + hueOffset;
+      this.saturation = LXUtils.clampf(LXColor.s(c) + saturationOffset, 0, 100);
```
Comment (zranger1) [inline src/main/java/titanicsend/lx/LXGradientUtils.java:125]: On line 124, does this
`this.hue = color.getHuef() + hueOffset;`
need to be range checked, or wrapped at 0 and 360, or is it done somewhere downstream?

**Review by zranger1 — CHANGES_REQUESTED** [review]
Big visual improvement!

File: src/main/java/titanicsend/lx/LXGradientUtils.java (hunk 1 above)
Comment (jkbelcher) [reply src/main/java/titanicsend/lx/LXGradientUtils.java:125]: Yeah it looks like it wants to be wrapped but this is a direct copy of LX's GradientUtils class and the two internal classes using it seem to be wrapping hue just fine.  It must be getting sorted out in the LXColor.hsb(...) call.
image

Comment (zranger1) [reply src/main/java/titanicsend/lx/LXGradientUtils.java:125]: Good enough!   I thought it'd be caught elsewhere in the framework, but just wanted to be sure.

**Pull request merged**